spec/draft/API_specification/indexing.rst

```diff
@@ -173,6 +179,34 @@ Multi-dimensional arrays must extend the concept of single-axis indexing to mult
 
     *Rationale: this is consistent with bounds-checking for single-axis indexing. An implementation may choose to set the axis (dimension) size of the result array to* ``0`` *, raise an exception, return junk values, or some other behavior depending on device requirements and performance considerations.*
 
+Integer Array Indexing
+----------------------
+
+An array must support indexing by an indexing tuple which contains only integers and integer arrays according to the following rules. Let ``A`` be an ``N``-dimensional array with shape ``S1``. Let ``T`` be a tuple ``(t1, t2, ..., tN)`` having length ``N``. Let ``tk`` be an individual element of ``T``.
```

Comment:
I wonder if it might be helpful to help with an informal description  followed by the details would make this section easier to read, e.g., "The array standard supported vectorized indexing by multi-dimensional integer arrays with broadcasting, as long as there is an indexing array for each dimension of the indexed array"

Reply:
I've added a note as a preface to this section.